modules/dxp/apps/search-experiences/search-experiences-test/src/testIntegration/java/com/liferay/search/experiences/internal/upgrade/test/SXPBlueprintUpgradeProcessTest.java

```diff
@@ -0,0 +1,480 @@
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Objects;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Almir Ferreira
+ */
+@RunWith(Arquillian.class)
+public class SXPBlueprintUpgradeProcessTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerMethodTestRule.INSTANCE);
+
+	@Before
+	public void setUp() throws Exception {
+		_group = GroupTestUtil.addGroup();
+		_user = TestPropsValues.getUser();
+
+		_serviceContext = ServiceContextTestUtil.getServiceContext(
+			_group, _user.getUserId());
+	}
+
+	@After
+	public void tearDown() throws Exception {
+		Iterator<Long> sxpBlueprintIdsIterator = _sxpBlueprintIds.iterator();
+
+		while (sxpBlueprintIdsIterator.hasNext()) {
+			_sxpBlueprintLocalService.deleteSXPBlueprint(
+				sxpBlueprintIdsIterator.next());
+
+			sxpBlueprintIdsIterator.remove();
+		}
+
+		Iterator<Long> sxpElementIdsIterator = _sxpElementIds.iterator();
+
+		while (sxpElementIdsIterator.hasNext()) {
+			_sxpElementLocalService.deleteSXPElement(
+				sxpElementIdsIterator.next());
+
+			sxpElementIdsIterator.remove();
+		}
+	}
+
+	@Test
+	public void testUpgradeEmptySXPBlueprintAndSXPElement() throws Exception {
+		SXPBlueprint sxpBlueprint = _addSXPBlueprint(null, null, null);
+		SXPElement sxpElement = _addSXPElement(null);
+
+		_runAllUpgrades();
+
+		_assertSXPBlueprint("[]", sxpBlueprint);
+		_assertSXPElement("", sxpElement);
+	}
+
+	@Test
+	public void testUpgradePortletPreferenceValue() throws Exception {
+		SXPBlueprint sxpBlueprint = _addSXPBlueprint(
+			_sxpBlueprintConfigurationJSONObject.toString(), null, null);
+
+		Long sxpBlueprintId = sxpBlueprint.getSXPBlueprintId();
+
+		PortletPreferenceValue searchBarSuggestions =
+			_addPortletPreferenceValue(
+				"suggestionsContributorConfigurations", _SEARCH_BAR_PORTLET_ID,
+				JSONUtil.put(
+					"attributes",
+					JSONUtil.put(
+						"includeAssetSearchSummary", true
+					).put(
+						"includeAssetURL", true
+					).put(
+						"sxpBlueprintId", sxpBlueprintId
+					)
+				).put(
+					"contributorName", "sxpBlueprint"
+				).put(
+					"displayGroupName", "suggestions"
+				).put(
+					"size", "5"
+				).toString(),
+				null);
+
+		PortletPreferenceValue lowLevelSearchOptions =
+			_addPortletPreferenceValue(
+				"attributes", _LOW_LEVEL_SEARCH_OPTIONS_PORTLET_ID, null,
+				"[{\"key\":\"search.experiences.blueprint.id\",\"value\":\"" +
+					sxpBlueprintId + "\"}]");
+
+		PortletPreferenceValue sxpBlueprintOptions = _addPortletPreferenceValue(
+			"sxpBlueprintId", _SXP_BLUEPRINT_OPTIONS_PORTLET_ID, null,
+			String.valueOf(sxpBlueprintId));
+
+		_addPortletPreferenceValue(
```

Comment:
why is it added?

Reply:
It's a portletPreferenceValue with a excluded blueprint